src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js

```diff
@@ -0,0 +1,90 @@
+(function (global, doc) {
+    const eZ = global.eZ = global.eZ || {};
+
+    eZ.baseFieldValidator = class BaseFieldValidator {
+        constructor(config) {
+            this.classInvalid = config.classInvalid;
+            this.eventsMap = config.eventsMap;
+            this.fieldSelector = config.fieldSelector;
+        }
+
+        attachEvent(eventName, selector, callback) {
+            [...doc.querySelectorAll(selector)].forEach(item => item.addEventListener(eventName, callback, false));
+        }
+
+        removeEvent(eventName, selector, callback) {
+            [...doc.querySelectorAll(selector)].forEach(item => item.removeEventListener(eventName, callback, false));
+        }
+
+        findStateNodes(fieldNode, input, selectors) {
+            return selectors.reduce((total, selector) => total.concat([...fieldNode.querySelectorAll(selector)]), []);
+        }
+
+        toggleInvalidState(isError, config, input) {
+            const fieldNode = input.closest(this.fieldSelector);
+            const methodName = isError ? 'add' : 'remove';
+            const stateNodes = this.findStateNodes(fieldNode, input, config.invalidStateSelectors);
+
+            fieldNode.classList[methodName](this.classInvalid);
+            input.classList[methodName](this.classInvalid);
+
+            stateNodes.forEach(el => el.classList[methodName](this.classInvalid));
+        }
+
+        createErrorNode(message) {
+            const errorNode = doc.createElement('em');
+
+            errorNode.classList.add('ez-field-error');
+            errorNode.innerHTML = message;
+
+            return errorNode;
+        }
+
+        findErrorNodes(fieldNode, input, selectors) {
+            return selectors.reduce((total, selector) => total.concat([...fieldNode.querySelectorAll(selector)]), []);
+        }
+
+        findExistingErrorNodes(fieldNode, input, selectors) {
```

Comment:
Is it needed? You could achieve the same by calling `this.findErrorNodes(fieldNode, input, selectors)`